internal/resources/connections/resource_metrics_endpoint_scrape_job.go

```diff
@@ -0,0 +1,134 @@
+package connections
+
+import (
+	"context"
+	"github.com/grafana/terraform-provider-grafana/v3/internal/common"
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64default"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"regexp"
+)
+
+var (
+	resourceMetricsEndpointScrapeJobTerraformName = "grafana_connections_metrics_endpoint_scrape_job"
+	resourceMetricsEndpointScrapeJobTerraformID   = common.NewResourceID(common.StringIDField("stack_id"), common.StringIDField("name"))
+)
+
+type resourceMetricsEndpointScrapeJob struct {
+}
+
+var Resources = makeResourceMetricsEndpointScrapeJob()
+
+func makeResourceMetricsEndpointScrapeJob() *common.Resource {
+	return common.NewResource(
+		common.CategoryConnections,
+		"grafana_cloud_provider_aws_cloudwatch_scrape_job",
+		resourceMetricsEndpointScrapeJobTerraformID,
+		&resourceMetricsEndpointScrapeJob{},
+	)
+}
+
+func (r resourceMetricsEndpointScrapeJob) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (r resourceMetricsEndpointScrapeJob) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = resourceMetricsEndpointScrapeJobTerraformName
+}
+
+var urlRegexp = regexp.MustCompile(`((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=\+\$,\w]+@)?[A-Za-z0-9.-]+(:[0-9]+)?|(?:www.|[-;:&=\+\$,\w]+@)[A-Za-z0-9.-]+)((?:\/[\+~%\/.\w-_]*)?\??(?:[-\+=&;%@.\w_]*)#?(?:[\w]*))?)`)
```

Comment:
Url regexp might be overkill, should we even validate it? I noticed it's not validated in the cloud-onboarding repo.

Reply:
Nevermind, the url is parsed in cloud-onboarding.

Reply:
Removing for now